### PR TITLE
EVG-2496 Add task group task setup and teardown

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

**Comment by bsamek** [comment]
The first commit has already been code reviewed. The second is the one that counts.

File: agent/agent_integration_test.go [hunk 1]
```diff
@@ -1,97 +0,0 @@
-package agent
```
Comment (bsamek) [inline agent/agent_integration_test.go]: I removed this because it's covered by TestAbort in agent_test.go

Comment (tychoish) [reply agent/agent_integration_test.go]: covered? or just duplicated?

Comment (bsamek) [reply agent/agent_integration_test.go]: Covered. The test in agent_test.go actually does slightly more than the integration test.

**Pull request merged**